app.js

```diff
@@ -1,16 +1,99 @@
-/* Imports */
+// function getRandomItem(array) {
+//     const random = getRandomNumber(array.length);
+//     const item = array[random];
+//     return item;
+// }
+
+function getRandomNumber(choices) {
+    return Math.floor(Math.random() * choices);
+}
+
+const opponents = ['boot', 'roach', 'nuke'];
 
 /* State */
+let gameState = 'pick'; // 'pick', or 'results'
+let pick = ''; // 'boot', 'roach', 'nuke'
+let opponentPick = ''; // 'boot', 'roach', 'nuke'
+// let result = ''; // 'win', 'lose', or 'draw'
 
 /* Actions */
-function loadPage() {}
+function loadPage() {
+    displayPick();
+    displayResults();
+}
 
-/* Components */
+function battle(userPick) {
+    gameState = 'results';
+    pick = userPick;
+    opponentPick = getRandomNumber(opponents);
 
+    if (pick === opponentPick) {
+        // result = 'draw';
+    } else if (pick === boot) {
+        if (opponentPick === 'roach') {
+            // result = 'win';
+        } else {
+            // result = 'loss';
+        }
+    } else if (pick === roach) {
```

Comment:
good logic for calculating the winner — you just need to figure out what to do with ```result``` (use ```result = 'win'``` to keep track of the state and use that state to re render what you want to show up when there's a win.

you'll also want to have wins / total in state so the ```result = 'win'``` also adds one to ```wins```. That way you can use the ```wins``` state to update the view